src/main/java/Duke.java

```diff
@@ -1,10 +1,91 @@
+import java.util.Arrays;
+import java.util.Scanner;
+
 public class Duke {
+    public static Task[] tasks = new Task[100];
+    public static int taskCount = 0;
+
+    public static void printGreeting() {
+        System.out.println("Hello! I'm Duke\n" + "What can I do for you?");
+    }
+    public static void printConfused() {
+        System.out.println("Could you say that again?");
+    }
+    public static void printExit() {
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+    public static void printGotIt() {
+        System.out.println("Got it. I've added this task:");
+    }
+    public static void printTaskCount() {
+        System.out.println("Now you have " + taskCount + " tasks in the list.");
+    }
+    public static void printTaskTypeResponse() {
+        //printing different responses depending if its Todo/Deadline/Event
+        printGotIt();
+        tasks[taskCount - 1].printTask();
+        printTaskCount();
+    }
+    public static void printList() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < taskCount; i += 1) {
+            System.out.print((i + 1) + ".");
+            tasks[i].printTask();
+        }
+    }
+    public static void TaskDone(int index) {
+        tasks[index].setDone();
+        System.out.println("Nice! I've marked this task as done:");
+        tasks[index].printTask();
+    }
+    public static void addTodo(String line) {
+        tasks[taskCount] = new Todo(line);
+        taskCount = taskCount + 1;
+        printTaskTypeResponse();
+    }
+    public static void addDeadline(String line) {
+        //extracting the description and date
+        String description = line.replaceAll("/.+", "");
+        String by = line.replaceAll(".+/by", "");
+        tasks[taskCount] = new Deadline(description, by);
+        taskCount = taskCount + 1;
+        printTaskTypeResponse();
+    }
+    public static void addEvent(String line) {
+        //extracting the description and date
+        String description = line.replaceAll("/.+", "");
+        String by = line.replaceAll(".+/at", "");
+        tasks[taskCount] = new Event(description, by);
+        taskCount = taskCount + 1;
+        printTaskTypeResponse();
+    }
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printGreeting();
+        String line;
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine();
+        //continue to run program unless types "bye" to exit program
+        while (!line.equals("bye")) {
```

Comment:
I do not really like the main function as it contains quite a few low level programming lines. Perhaps you could put them inside a separate method and then call this method in the main function.